shell.nix

```diff
@@ -0,0 +1,23 @@
+{ pkgs ? import <nixpkgs> {
+    overlays = [
+(import "${(fetchTarball "https://github.com/nix-community/fenix/archive/main.tar.gz")}/overlay.nix")
+      (self: super: {
+          rustc = super.fenix.latest.rustc;
+          cargo  = super.fenix.latest.cargo;
+          rust-src = super.fenix.latest.rust-src;
+      }
+        )
+    ];
+  }
+}:
+
+pkgs.mkShell {
+  packages = with pkgs; [
+    rustc
+    cargo
+    rust-analyzer
+  ];
```

Comment:
I wonder if this should include packages needed to run the integration test (at least bitcoin and clightning). Probably more useful.